backend/database/migrations/2021_01_22_061707_allow_user_name_null.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+
+class AllowUserNameNull extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::table('users', function (Blueprint $table) {
+            $table->string('name')->nullable()->change();
+        });
+    }
+
+    /**
+     * Reverse the migrations.
+     *
+     * @return void
+     */
+    public function down()
+    {
+        Schema::table('users', function (Blueprint $table) {
+            $table->string('name')->nullable(false)->change();
```

Comment:
How does this work? 

If the `down()` methods of all the migrations are ran chronologically, the `users` table won't exist to be modified by the time this `down()` method is ran. Doesn't that make this redundant?

Or is it possible that this `down()` method can be ran independent of the others? (new to me)

Reply:
The down methods would be run in reverse chronological order. It's the undo of up. Also you can just undo a single migration (artisan migrate:rollback --step=1)